src/Makefile

```diff
@@ -1325,6 +1326,39 @@ external/$(CPYTHON).tar.gz:
 	rm $(patsubst %.gz,%,$@)
 endif
 
+ifeq (${uname_S},HP-UX)
+# nlohmann library
+RESOURCES_URL_BASE_NLOHMANN=$(subst https://,http://,$(RESOURCES_URL_BASE))
+ifeq ($(findstring $(RESOURCES_URL_BASE_NLOHMANN),$(RESOURCES_URL)),$(RESOURCES_URL_BASE_NLOHMANN))
```

Comment:
Just a detail, no changes are requested: if the `RESOURCES_URL` variable is externally set with `https` instead of `http`, the mechanism of setting the **nlohmann** library to version 21 won't be applied.

But this doesn't seem a problem, because in HP-UX, we always should define it with `http`.